packer/builder/azure/config.go

```diff
@@ -129,6 +131,20 @@ func newConfig(raws ...interface{}) (*Config, []string, error) {
 		errs = packer.MultiErrorAppend(errs, fmt.Errorf("instance_size is not valid, must be one of: %v", allowedVMSizes))
 	}
 
+	for n := 0; n < len(c.DataDisks); n++ {
+		switch v := c.DataDisks[n].(type) {
+		case string:
```

Comment:
this looks too much defensive coding. there's no way to get a single input here? 

looks like you're also discarding the `v` if it's string/int?